### PR TITLE
Release 7.1.0

### PR DESCRIPTION
**Pull request by JinmingHu-MSFT**
1. Add support for user delegation SAS
2. Fix a bug in snapshot SAS: SAS token should not contain snapshot time query parameter
3. Fix some failed ut cases.
4. Add an API with which user can custom HTTP connection settings.
4. Remove compile options -fprofile-arcs -ftest-coverage

### CODE REVIEW

File: Microsoft.WindowsAzure.Storage/tests/cloud_storage_account_test.cpp [hunk 1]
```diff
@@ -920,10 +922,15 @@ SUITE(Core)
             policy.set_resource_type((azure::storage::account_shared_access_policy::resource_types)0x7);
 
             policy.set_permissions((uint8_t)i);
+            std::cout << "i == " + std::to_string(i) + " here1\n" << std::flush;
```
Comment (katmsft) [inline Microsoft.WindowsAzure.Storage/tests/cloud_storage_account_test.cpp]: >std::cout << "i == " + std::to_string(i) + " here1\n" << std::flush; [](start = 12, length = 68)

I believe you forgot to remove the debug information.

Comment (JinmingHu-MSFT) [reply Microsoft.WindowsAzure.Storage/tests/cloud_storage_account_test.cpp]: sure

**Comment by JinmingHu-MSFT** [comment]
@katmsft Can you take time to review the last commit https://github.com/Azure/azure-storage-cpp/pull/310/commits/cd19106fb2437f2c8e07bdb5e556320dca2f55f2?

**Comment by JinmingHu-MSFT** [comment]
@katmsft Can you take time to review the last 4 commits, especially https://github.com/Azure/azure-storage-cpp/pull/310/commits/e73aae85af923996bb57953e4f3949fc98c0ebb7?

File: Microsoft.WindowsAzure.Storage/src/cloud_blob.cpp [hunk 2]
```diff
@@ -803,15 +828,13 @@ namespace azure { namespace storage {
...
-                                        {
-                                            semaphore->unlock();
-                                        }
                                     }
                                 }
                             });
                         });
+                        parallel_tasks.emplace_back(std::move(parallel_task));
                     }
+                    // Code below is nonsense, becasuse exceptions won't be thrown from wait_all_async.
```
Comment (katmsft) [inline Microsoft.WindowsAzure.Storage/src/cloud_blob.cpp]: Then I think it is proper to delete these code.

Comment (JinmingHu-MSFT) [reply Microsoft.WindowsAzure.Storage/src/cloud_blob.cpp]: Yeah, but that is legacy code from ancient times, dare we just delete it?

Comment (katmsft) [reply Microsoft.WindowsAzure.Storage/src/cloud_blob.cpp]: I think a comment like this here is indicating that the code is useless, hence we should be able to safely delete it.

Comment (JinmingHu-MSFT) [reply Microsoft.WindowsAzure.Storage/src/cloud_blob.cpp]: I think it indicates that the way exceptions being handled is wrong, but doesn't in any sense suggest the semaphore operation is useless.

Comment (JinmingHu-MSFT) [reply Microsoft.WindowsAzure.Storage/src/cloud_blob.cpp]: If it bothers you that much, I'll just delete it.

Comment (JinmingHu-MSFT) [reply Microsoft.WindowsAzure.Storage/src/cloud_blob.cpp]: Done

**Comment by JinmingHu-MSFT** [comment]
@katmsft Can you take time to review the last 3 commits?

**Pull request merged**